### PR TITLE
WRR-12139: Convert sandstone/ThemeDecorator/I18nFontDecorator to functional component

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Changed I18NFontDecorator to be a functional component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-12139

### CODE REVIEW

File: ThemeDecorator/I18nFontDecorator.js [hunk 1]
```diff
@@ -1,47 +1,25 @@
...
+		return <Wrapped {...props} />;
+	};
 
-		render () {
-			return (
-				<Wrapped {...this.props} />
-			);
-		}
+	I18nDecorator.displayName = 'I18nFontDecorator';
+	I18nDecorator.propTypes = {
```
Comment (daniel-stoian-lgp) [inline ThemeDecorator/I18nFontDecorator.js:18]:
```suggestion
	I18nDecorator.displayName = 'I18nFontDecorator';
	
	I18nDecorator.propTypes = {
```

File: ThemeDecorator/I18nFontDecorator.js [hunk 2]
```diff
@@ -1,47 +1,25 @@
...
 
 import {fontOverrideGenerator} from './fontGenerator';
 
 const I18nFontDecorator = hoc((config, Wrapped) => {
-	return class I18nDecorator extends Component {
-		static displayName = 'I18nFontDecorator';
+	const I18nDecorator = (props) => {
+		useEffect(() => {
+			fontOverrideGenerator(props.locale || ilib.getLocale());
+		}, [props.locale]);
```
Comment (daniel-stoian-lgp) [inline ThemeDecorator/I18nFontDecorator.js:12]: this useEffect() is not called whenever the locale changes in storybook, while in the previous implementation, componentDidUpdate() was called .
Please fix

File: ThemeDecorator/I18nFontDecorator.js [hunk 3]
```diff
@@ -1,47 +1,25 @@
 import hoc from '@enact/core/hoc';
 import ilib from '@enact/i18n';
-import {Component} from 'react';
+import {useEffect} from 'react';
 import PropTypes from 'prop-types';
```
Comment (daniel-stoian-lgp) [inline ThemeDecorator/I18nFontDecorator.js:4]: sort imports, please

File: ThemeDecorator/I18nFontDecorator.js (hunk 1 above)
Comment (alexandrumorariu) [reply ThemeDecorator/I18nFontDecorator.js]: Done

File: ThemeDecorator/I18nFontDecorator.js (hunk 2 above)
Comment (alexandrumorariu) [reply ThemeDecorator/I18nFontDecorator.js]: Done

File: ThemeDecorator/I18nFontDecorator.js (hunk 3 above)
Comment (alexandrumorariu) [reply ThemeDecorator/I18nFontDecorator.js]: Done

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**